hackathon_site/hardware/serializers.py

```diff
@@ -0,0 +1,41 @@
+from rest_framework import serializers
+from hardware.models import Hardware, Category, Order
+
+
+class HardwareSerializer(serializers.ModelSerializer):
+    quantity_remaining = serializers.SerializerMethodField()
+
+    class Meta:
+        model = Hardware
+        fields = (
+            "id",
+            "name",
+            "model_number",
+            "manufacturer",
+            "datasheet",
+            "quantity_available",
+            "notes",
+            "max_per_team",
+            "picture",
+            "categories",
+            "quantity_remaining",
+        )
+
+    def get_quantity_remaining(self, obj):
+        return (
+            obj.quantity_available
+            - Order.objects.filter(items__hardware__id=obj.id).count()
```

Comment:
This isn't entirely correct, because:
- We should be counting items that have been returned as available again
- When items are just in their cart, they have not been checked out yet and are still available for other teams to snag. We decided on this so people can't just hoard things in their cart.